### DISCUSSION TITLE
2023-03 hovers no longer work

### DISCUSSION THREAD

**Discussion by scsirob** [eclipse-cdt/cdt#338] [Q&A] [answered]
Not sure if anyone else has this, but all hovers stopped working for me on 2023-03 C/C++ CDT. Even a clean fresh installation does not show any values, man pages or routine layouts when hovering over items. I checked hover settings and scalability settings, all seem in order.
For my main dev system I reverted back to 2022-12 and hovers work again. Anyone else seeing this?

**Comment by jonahgraham** [comment]
Can you expand on what platform you are running on that doesn't work. 

This is what I see on 2023-03 running on Linux:

![Peek 2023-06-07 15-11](https://github.com/eclipse-cdt/cdt/assets/679236/e9961a6e-bf35-4e26-8608-254b03a0a2ca)

**Reply by scsirob** [reply]
Thanks for reviewing the issue. On 2022-12 I see the correct behaviour as displayed in your example. On 2023-03 that functionality was completely gone. None of the hovers worked. I use the Darkest dark theme from Genuitec.

I run eclipse on an Ubuntu 20.04.6 LTS VM and openjdk version "11.0.19" 2023-04-18, with output going to vcXsrv 1.20.14.0 on a Windows-10 desktop system. Login from Windows to Linux over ssh using PuTTY with X11 Forwarding enabled.

**Reply by jonahgraham** [reply]
I guess something may have changed in vnc handling, but not sure where to start with that. CDT itself relies on mostly standard Eclipse platform ways of displaying such pop-ups. In the screenshot I shared there were three types of hovers, two done by CDT and one from [Linux Tools](https://github.com/eclipse-linuxtools/org.eclipse.linuxtools) (IIRC). 

If you can spend some more time trying to reproduce this, you can try downloading the Eclipse SDK from https://download.eclipse.org/eclipse/downloads/drops4/R-4.27-202303020300/ ([direct link to linux tar](https://download.eclipse.org/eclipse/downloads/drops4/R-4.27-202303020300/download.php?dropFile=eclipse-SDK-4.27-linux-gtk-x86_64.tar.gz)) and make a Java project to see if the hover pop-ups work there. 

Eclipse 4.27 is the base for Eclipse 2023-03 - [Eclipse 4.26](https://archive.eclipse.org/eclipse/downloads/drops4/R-4.26-202211231800/) is the base for Eclipse 2022-12


> openjdk version "11.0.19" 

BTW CDT requires Java 17, my guess is that Eclipse isn't using java from your PATH, but rather a bundled Java. You can see which java is in use from Help -> About -> Installation Details -> Configuration tab -> java.home setting.

PS Are you using MyEclipse from Genuitec? I recommend filing the issue with them.

**Reply by scsirob** [reply]
You are correct on the Java version, java.home.settings uses the bundled Java 17.
The install was a direct download from the eclipse.org site, which was upgraded multiple times as time went by. The installation history shows 4.25.0.20220908-1200 as the initial install, with two version upgrades, and now a revert applied. The Genuintec theme was applied through the marketplace.
I have installed the Eclipse SDK from your link on the same VM and it does provide hovers in Java.

**Reply by jonahgraham** [reply, marked as answer]
I tried:

1. installing Eclipse for C/C++ Devs 2022-09, upgrading it to 2022-12 and then 2023-03. Hovers still worked 
2. Run Eclispe through Tiger VNC (the one that comes with Ubuntu 20.04 IIRC - `Xvnc TigerVNC 1.7.0`) and viewer `TigerVNC Viewer 64-bit v1.10.1` - all running on same machine. Hovers still work  
3. Installed Genuitec darkest dark from Marketplace (still in VNC). Hovers still worked

I don't know what to try next to reproduce the problem. It is also strange that Eclipse SDK worked fine, but CDT didn't. I wonder if something has been corrupted in the installation because of multiple upgrades. 

If you can create a reproducible case then I can investigate further.

**Comment by scsirob** [comment]
Thanks for following up. I guess it's a corner case that's very hard to reproduce. Not really worth it.. Go ahead and close this case, I'll re-install from scratch and if that doesn't fix things I'll re-report.

**Reply by jonahgraham** [reply]
Sounds good!
